docs/usage/guides/cuda.md

```diff
@@ -11,53 +11,125 @@ To get the NVIDIA container runtime in the K3S image you need to build your own
 
-### Adapt the Dockerfile
+### Dockerfiles:  
+  
+Dockerfile.base:
+```Dockerfile
+FROM nvidia/cuda:11.2.0-base-ubuntu18.04
+
+ENV DEBIAN_FRONTEND noninteractive
+
+ARG DOCKER_VERSION
+ENV DOCKER_VERSION=$DOCKER_VERSION
+
+RUN set -x && \
+    apt-get update && \
+    apt-get install -y \
+    apt-transport-https \
+    ca-certificates \
+    curl \
+    wget \
+    tar \
+    zstd \
+    gnupg \
+    lsb-release \
+    git \
+    software-properties-common \
+    build-essential && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN set -x && \
+    curl -fsSL https://download.docker.com/linux/$(lsb_release -is | tr '[:upper:]' '[:lower:]')/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg && \
+    echo "deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/$(lsb_release -is | tr '[:upper:]' '[:lower:]') $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null && \
+    apt-get update && \
+    apt-get install -y \
+    containerd.io \
+    docker-ce=5:$DOCKER_VERSION~3-0~$(lsb_release -is | tr '[:upper:]' '[:lower:]')-$(lsb_release -cs) \
+    docker-ce-cli=5:$DOCKER_VERSION~3-0~$(lsb_release -is | tr '[:upper:]' '[:lower:]')-$(lsb_release -cs) && \
+    rm -rf /var/lib/apt/lists/*
+
+```  
+  
+  
+  
+Dockerfile.k3d-gpu:  
 
 ```Dockerfile
-FROM ubuntu:18.04 as base
-RUN apt-get update -y && apt-get install -y ca-certificates
-ADD k3s/build/out/data.tar.gz /image
-RUN mkdir -p /image/etc/ssl/certs /image/run /image/var/run /image/tmp /image/lib/modules /image/lib/firmware && \
+FROM nvidia/cuda:11.2.0-base-ubuntu18.04 as base
+
+RUN set -x && \
+    apt-get update && \
+    apt-get install -y ca-certificates zstd
+
+COPY k3s/build/out/data.tar.zst /
+
+RUN set -x && \
+    mkdir -p /image/etc/ssl/certs /image/run /image/var/run /image/tmp /image/lib/modules /image/lib/firmware && \
+    tar -I zstd -xf /data.tar.zst -C /image && \
     cp /etc/ssl/certs/ca-certificates.crt /image/etc/ssl/certs/ca-certificates.crt
-RUN cd image/bin && \
+
+RUN set -x && \
+    cd image/bin && \
     rm -f k3s && \
     ln -s k3s-server k3s
 
-FROM ubuntu:18.04
-RUN echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections
-RUN apt-get update -y && apt-get -y install gnupg2 curl
+FROM nvidia/cuda:11.2.0-base-ubuntu18.04
+
+RUN set -x && \
+    echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections
+
+RUN set -x && \
+    apt-get update && \
+    apt-get -y install gnupg2 curl
+
+# Install NVIDIA Container Runtime
+RUN set -x && \
+    curl -s -L https://nvidia.github.io/nvidia-container-runtime/gpgkey | apt-key add -
+
+RUN set -x && \
+    curl -s -L https://nvidia.github.io/nvidia-container-runtime/ubuntu18.04/nvidia-container-runtime.list | tee /etc/apt/sources.list.d/nvidia-container-runtime.list
+
+RUN set -x && \
+    apt-get update && \
+    apt-get -y install nvidia-container-runtime=3.5.0-1
```

Comment:
Maybe we could make this version configurable via an Dockerfile `ARG`?

Reply:
Sure. Configured it using ARG/ENV + added a related export to the bash script